internal/security/bootstrapper/tcp/listener_test.go

```diff
@@ -0,0 +1,104 @@
+/*******************************************************************************
+ * Copyright 2021 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ *******************************************************************************/
+
+package tcp
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+	"github.com/stretchr/testify/require"
+)
+
+func TestStartListener(t *testing.T) {
+	lc := logger.MockLogger{}
+	errs := make(chan error, 1)
+	testPort := 12345
+	srv := NewTcpServer()
+	go func() {
+		errs <- srv.StartListener(testPort, lc, "")
+	}()
+
+	go func() {
+		errs <- DialTcp("127.0.0.1", testPort, lc)
+	}()
+
+	// give some time for connection
+	time.Sleep(3 * time.Second)
+
+	require.NoError(t, <-errs)
+	srv.lock.Lock()
+	defer srv.lock.Unlock()
+	accepted := srv.accepted
+	require.True(t, accepted)
+
+}
+
+func TestStartListenerAlreadyInUse(t *testing.T) {
+	lc := logger.MockLogger{}
+	errs := make(chan error, 1)
+	testPort := 12347
+	srv1 := NewTcpServer()
+	go func() {
+		errs <- srv1.StartListener(testPort, lc, "")
+	}()
+
+	// try to start another listener with the same port
+	// this will cause an error
+	srv2 := NewTcpServer()
+	go func() {
+		errs <- srv2.StartListener(testPort, lc, "")
+	}()
+
+	time.Sleep(time.Second)
+
+	go func() {
+		errs <- DialTcp("127.0.0.1", testPort, lc)
+	}()
+
+	// give some time for connection
+	time.Sleep(time.Second)
+
+	require.Error(t, <-errs)
+}
+
+func TestStartListenerServerNotReady(t *testing.T) {
+	lc := logger.MockLogger{}
+	errs := make(chan error, 1)
+	testPort := 12349
+	go func() {
+		errs <- DialTcp("127.0.0.1", testPort, lc)
+	}()
+
+	time.Sleep(dialTimeoutDuration + time.Second)
+
+	// since the tcp server is never up, the dial will block forever,
+	// so we set a timeout for the test to signal it is done
+	timeout := make(chan bool, 1)
+	go func() {
+		time.Sleep(2 * time.Second)
+		timeout <- true
+	}()
```

Comment:
better wat to time out is in example in previous comment

Reply:
changed